upup/pkg/kutil/rollingupdate_cluster.go

```diff
@@ -239,25 +246,42 @@ func buildCloudInstanceGroup(ig *api.InstanceGroup, g *autoscaling.Group, nodeMa
 	return n
 }
 
-func (n *CloudInstanceGroup) RollingUpdate(cloud fi.Cloud, force bool, interval time.Duration, k8sClient *release_1_5.Clientset) error {
+// Performs a rolling update on a list of nodes
+func (n *CloudInstanceGroup) RollingUpdate(cloud fi.Cloud, force bool, interval time.Duration, instanceGroupList *api.InstanceGroupList, k8sClient *release_1_5.Clientset) error {
 	c := cloud.(awsup.AWSCloud)
 
 	update := n.NeedUpdate
 	if force {
 		update = append(update, n.Ready...)
 	}
+
 	for _, u := range update {
-		instanceID := aws.StringValue(u.ASGInstance.InstanceId)
-		glog.Infof("Stopping instance %q in AWS ASG %q", instanceID, n.ASGName)
 
-		// TODO: Evacuate through k8s first?
+		drain, drainErr := NewDrainOptions(nil, u.Node.ClusterName)
 
-		// TODO: Temporarily increase size of ASG?
+		if drainErr != nil {
+			glog.Warningf("Error creating drain: %v", drainErr)
+		} else {
+			drainErr = drain.SetupDrain(u.Node.Name)
+			if drainErr != nil {
+				glog.Warningf("setupErr: %v", drainErr)
+			} else {
+				drainErr = drain.RunDrain()
+
+				if drainErr != nil {
+					// TODO: fail here??
+					glog.Warningf("Drain failed %s", u.Node.Name)
```

Comment:
I don't the consequence of a drain failing - it sounds like the entire operation should fail? I don't know - concurrent failures are terrifying to be honest - I would need to know more about what we are attempting to do here, and what `drain` means in k8s (to you)